python/pyspark/ml/base.py

```diff
@@ -59,6 +59,12 @@ def fit(self, dataset, params=None):
             return [self.fit(dataset, paramMap) for paramMap in params]
         elif isinstance(params, dict):
             if params:
+                if isinstance(params.keys()[0],str):
+                    param_new = dict()
+                    for param, value in params.items():
+                        p = getattr(self, param)
+                        param_new.update({p: value})
+                    params=param_new
```

Comment:
Could we just make it just like the one as below?

```python
params = dict([(p, v) for p, v in params.items()])
```

BTW, I am not supposed to decide what should be added into Spark. We can wait for committer's review :)